src/main/java/me/botsko/prism/commands/ReportCommand.java

```diff
@@ -12,273 +9,233 @@
+        } else {
+            sender.sendMessage(
+                    Prism.messenger.playerError("Recorder stopped running! DB conn problems? Try /pr recorder start"));
+        }
+
+        sender.sendMessage(Prism.messenger.playerSubduedHeaderMsg("Attempting to check connection readiness..."));
+
+        Connection conn = null;
+        try {
+
+            conn = Prism.getPrismDataSource().getConnection();
+            if (conn == null) {
+                sender.sendMessage(Prism.messenger.playerError("Pool returned NULL instead of a valid connection."));
+            } else if (conn.isClosed()) {
+                sender.sendMessage(Prism.messenger.playerError("Pool returned an already closed connection."));
+            } else if (conn.isValid(5)) {
+                sender.sendMessage(Prism.messenger.playerSuccess("Pool returned valid connection!"));
+            }
+        } catch (final SQLException e) {
+            sender.sendMessage(Prism.messenger.playerError("Error: " + e.getMessage()));
+            e.printStackTrace();
+        } finally {
+            if (conn != null)
+                try {
+                    conn.close();
+                } catch (final SQLException ignored) {
+                }
+        }
+    }
+
+
+    private void blockSumReports(final CallInfo call) {
+
+        // Process and validate all of the arguments
+        final QueryParameters parameters = PreprocessArgs.process(plugin, call.getSender(), call.getArgs(),
+                PrismProcessType.LOOKUP, 3, !plugin.getConfig().getBoolean("prism.queries.never-use-defaults"));
+        if (parameters == null) {
+            call.getSender()
+                    .sendMessage(Prism.messenger.playerError("You must specify parameters, at least one player."));
+            return;
+        }
+        // No actions
+        if(checkParams(parameters,call)){
+            return;
+        }
+
+        final BlockReportQuery reportQuery = Prism.getPrismDataSource().createBlockReportQuery();
+        /*
+          Run the lookup itself in an async task so the lookup query isn't done on the
+          main thread
+         */
+        plugin.getServer().getScheduler().runTaskAsynchronously(plugin, () -> reportQuery.report(call.getSender()));
+    }
+    private boolean checkParams(QueryParameters parameters, CallInfo call){
+        if (!parameters.getActionTypes().isEmpty()) {
+            call.getSender()
+                    .sendMessage(Prism.messenger.playerError("You may not specify any action types for this report."));
+            return true;
+        }
+        // Verify single player name for now
+        final Map<String, MatchRule> players = parameters.getPlayerNames();
+        if (players.size() != 1) {
+            call.getSender().sendMessage(Prism.messenger.playerError("You must provide only a single player name."));
+            return true ;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [';' is preceded with whitespace.](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5438808)